topics/basics/testing_plugins/integration_tests/integration_tests_intro.md

```diff
@@ -0,0 +1,286 @@
+>
+{style="note"}
+
+## Understanding the Test Architecture
+
+The test can now be run and it should pass.
+
+Integration tests operate across two separate processes:
+
+* Test process:
+    * Executes test code, sending commands to the IDE.
+    * Manages the IDE life cycle.
+    * Controls test flow and assertions.
+* IDE process:
+    * Listens and executes commands from the test process.
+
+This dual-process architecture explains several key aspects of integration testing:
+
+* Why debugging requires special considerations.
+* The need for a communication protocol between test and IDE processes.
+* Why a built plugin distribution is required.
+* The origin of certain test-specific exceptions.
+
+## Opening Projects in Tests
+
+While starting an IDE with an empty project is useful, sometimes we need actual projects to verify real-world scenarios. Let's modify our test to open a project.
+
+The framework supports several ways to specify test projects:
+
+* **GitHub projects**: `GitHubProject.fromGithub( branchName = "master", "JetBrains/ij-perf-report-aggregator" )`
+* **Remote archives**: `RemoteArchiveProjectInfo("https://github.com/JetBrains/intellij-community/archive/master.zip")`
+* **Local projects**: `LocalProjectInfo(Path("src/test/resources/test-projects/simple-project"))`
+
+Here's our test modified to open a GitHub project:
+
+```kotlin
+@Test
+fun simpleTest() {
+  Starter.newContext(
+    "testExample",
+    TestCase(
+      IdeProductProvider.IC,
+      GitHubProject.fromGithub(branchName = "master", repoRelativeUrl = "JetBrains/ij-perf-report-aggregator")
+    ).withVersion("2024.2")
+  ).apply {
+    val pathToPlugin = System.getProperty("path.to.build.plugin")
+    PluginConfigurator(this).installPluginFromPath(Path(pathToPlugin))
+  }.runIdeWithDriver().useDriverAndCloseIde {
+    waitForIndicators(5.minutes)
+  }
+}
+```
+
+While simple, this test verifies a critical aspect: our plugin doesn't interfere with IDE startup.
+
+> We added a call `waitForIndicators` into the lambda to make sure that we wait till all indicators are gone before exiting the IDE.
+>
+{style="note"}
+
+## Catching Exceptions from IDE
+
+Our test has one critical limitation: it won't detect exceptions or freezes occurring within the IDE process. Let's understand why and how to fix this.
+
+Due to the two-process architecture:
+
+* Exceptions in the IDE process aren't automatically propagated to the test process.
+* A bundled plugin collects exceptions from the IDE's `MessageBus`.
+* These exceptions are stored in the error folder within the logs.
+
+The Starter framework uses TeamCity reporting by default, falling back to `NoCIServer` for other environments. However, we can customize this behavior using Kodein Dependency Injection. Here's how to make tests fail when IDE exceptions occur:
```

Comment:
```suggestion
The Starter framework uses TeamCity reporting by default, falling back to `NoCIServer` for other environments. 
However, this can be customized by using Kodein Dependency Injection.
The following example shows how to make tests fail when IDE exceptions occur:
```